toolchain/check/convert.cpp

```diff
@@ -833,6 +833,19 @@ static auto PerformBuiltinConversion(Context& context, Parse::NodeId parse_node,
         struct_literal->elements_id == SemIR::InstBlockId::Empty) {
       value_id = sem_ir.types().GetInstId(value_type_id);
     }
+
+    // Facet type conversions: a value T of facet type F1 can be implicitly
+    // converted to facet type F2 if T satisfies the requirements of F2.
+    //
+    // TODO: Support this conversion in general. For now we only support it in
+    // the case where F1 is an interface type and F2 is `type`.
+    // TODO: Support converting tuple and struct values to facet types,
+    // combining the above conversions and this one in a single conversion.
+    if (auto src_interface_type =
+            sem_ir.types().TryGetAs<SemIR::InterfaceType>(value_type_id)) {
```

Comment:
Why does this do TryGetAs when it doesn't use src_interface_type, vs Is? Is this about the TODOs? Still looks kind of odd to give it a name.

```suggestion
    if (sem_ir.types().TryGetAs<SemIR::InterfaceType>(value_type_id)) {
```

```suggestion
    if (sem_ir.types().GetAsInst(value_type_id).Is<SemIR::InterfaceType>()) {
```

Reply:
Added `TypeStore::Is`.